custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/src/main/scala/com/datarobot/custom/H2OPredictor.scala

```diff
@@ -0,0 +1,199 @@
+
+    val csvPrinter: CSVPrinter = this.isRegression match {
+      case true =>
+        new CSVPrinter(
+          new StringWriter(),
+          CSVFormat.DEFAULT.withHeader("Predictions")
+        )
+      case false =>
+        new CSVPrinter(
+          new StringWriter(),
+          CSVFormat.DEFAULT
+            .withHeader(this.positiveClassLabel, this.negativeClassLabel)
+        )
+    }
+
+    val predictions = Try(this.scoreStringCSV(stringCSV))
+
+    predictions match {
+      case Success(preds) => {
+        isRegression match {
+          case true =>
+            preds.foreach(p =>
+              csvPrinter.printRecord(p(0).asInstanceOf[Object])
+            )
+          case false =>
+            preds.foreach { p =>
+              csvPrinter.printRecord(
+                p(1).asInstanceOf[Object],
+                p(0).asInstanceOf[Object]
+              )
+            }
+        }
+      }
+      case Failure(e) => throw new Exception(e)
+    }
+
+    csvPrinter.flush()
+    val outStream: StringWriter =
+      csvPrinter.getOut().asInstanceOf[StringWriter];
+    outStream.toString()
+
+  }
+
+  def scoreStringCSV(stringCSV: String) = {
+
+    val csvFormat = CSVFormat.DEFAULT.withHeader();
+
+    val parser =
+      csvFormat.parse(new BufferedReader(new StringReader(stringCSV)))
+
+    val sParser = parser.iterator.asScala.map { _.toMap }.map { map2RowData }
+
+    val predictions = sParser.map { record =>
+      val prediction = this.isRegression match {
+        case true  => Array(this.model.predictRegression(record).value)
+        case false => this.model.predictBinomial(record).classProbabilities
+      }
+      prediction
+    }.toArray
+    predictions
+  }
+
+  def map2RowData(x: java.util.Map[String, String]): RowData = {
+    val row = x.asScala
+    val rowData: RowData = new RowData();
+    row.foreach { case (k, v) => rowData.put(k, v.toString) }
+    rowData
+  }
+
+  def modelConfigViaMojo(mojo: java.io.File): EasyPredictModelWrapper.Config = {
+    new EasyPredictModelWrapper.Config()
+      .setModel(MojoModel.load(mojo.getAbsolutePath))
+      .setConvertUnknownCategoricalLevelsToNa(true)
+      .setConvertInvalidNumbersToNa(true)
+  }
+
+  // this would work with a properly compiled / packaged pojo
+  // this will work with a compile pojo
+  // the expectation is that we compile the pojo when the container is spun up.
+// s"javac -cp code/h2o-genmodel-3.30.0.6.jar ${pojo.getAbsolutePath}".!
+
+  def modelConfigViaPojo(
+      pojo: java.io.File
+  ): EasyPredictModelWrapper.Config = {
+    val urls = pojo.getName.endsWith("java") match {
+      case true => {
+        val pathOfJar = classOf[hex.genmodel.GenModel].getProtectionDomain.getCodeSource.getLocation.toURI.getPath
+        s"javac -cp ${pathOfJar} ${pojo.getAbsolutePath}".!
+        Array(pojo.getAbsoluteFile.getParentFile.toURI.toURL)
+      }
+      case false => throw new Exception("Pojo not availabe")
```

Comment:
Exception veribage please

Reply:
I changed the way this method works which i think lends transparency to potential exceptions.  As the java file is compiled, if it exists with non zero exit code, exception is tossed and gives tips about what could be result.  

`s"executing '${compilePojo}' exited with non-zero code.  Make sure POJO exists and hasn't been renamed.  If the POJO is > 1GB in size, consider using MOJO")`